ssl/quic/qlog_event_helpers.c

```diff
@@ -0,0 +1,633 @@
+                goto unknown;
+
+            QLOG_STR("frame_type", "new_connection_id");
+            QLOG_U64("sequence_number", f.seq_num);
+            QLOG_U64("retire_prior_to", f.retire_prior_to);
+            QLOG_CID("connection_id", &f.conn_id);
+            QLOG_BIN("stateless_reset_token",
+                     f.stateless_reset.token,
+                     sizeof(f.stateless_reset.token));
+        }
+        break;
+    case OSSL_QUIC_FRAME_TYPE_RETIRE_CONN_ID:
+        {
+            uint64_t seq_num;
+
+            if (!ossl_quic_wire_decode_frame_retire_conn_id(pkt, &seq_num))
+                goto unknown;
+
+            QLOG_STR("frame_type", "retire_connection_id");
+            QLOG_U64("sequence_number", seq_num);
+        }
+        break;
+    case OSSL_QUIC_FRAME_TYPE_DATA_BLOCKED:
+        {
+            uint64_t x;
+
+            if (!ossl_quic_wire_decode_frame_data_blocked(pkt, &x))
+                goto unknown;
+
+            QLOG_STR("frame_type", "data_blocked");
+            QLOG_U64("limit", x);
+        }
+        break;
+    case OSSL_QUIC_FRAME_TYPE_STREAM_DATA_BLOCKED:
+        {
+            uint64_t stream_id, x;
+
+            if (!ossl_quic_wire_decode_frame_stream_data_blocked(pkt,
+                                                                 &stream_id,
+                                                                 &x))
+                goto unknown;
+
+            QLOG_STR("frame_type", "stream_data_blocked");
+            QLOG_U64("stream_id", stream_id);
+            QLOG_U64("limit", x);
+        }
+        break;
+    case OSSL_QUIC_FRAME_TYPE_STREAMS_BLOCKED_BIDI:
+    case OSSL_QUIC_FRAME_TYPE_STREAMS_BLOCKED_UNI:
+        {
+            uint64_t x;
+
+            if (!ossl_quic_wire_decode_frame_streams_blocked(pkt, &x))
+                goto unknown;
+
+            QLOG_STR("frame_type", "streams_blocked");
+            QLOG_STR("stream_type",
+                     frame_type == OSSL_QUIC_FRAME_TYPE_STREAMS_BLOCKED_BIDI
+                        ? "bidirectional" : "unidirectional");
+            QLOG_U64("limit", x);
+        }
+        break;
+    case OSSL_QUIC_FRAME_TYPE_NEW_TOKEN:
+        {
+            const unsigned char *token;
+            size_t token_len;
+
+            if (!ossl_quic_wire_decode_frame_new_token(pkt, &token, &token_len))
+                goto unknown;
+
+            QLOG_STR("frame_type", "new_token");
+            QLOG_BEGIN("token");
+                QLOG_BEGIN("raw");
+                    QLOG_BIN("data", token, token_len);
+                QLOG_END();
+            QLOG_END();
+        }
+        break;
+    default:
+unknown:
+        QLOG_STR("frame_type", "unknown");
+        QLOG_U64("frame_type_value", frame_type);
+
+        /*
+         * Can't continue scanning for frames in this case as the frame length
+         * is unknown. We log the entire body the rest of the packet payload
+         * as the raw data of the frame.
```

Comment:
This last sentence doesn't quite scan...is it meant to say "of the rest of"?

Reply:
Fixed